irodsadmin/management/commands/irodsorphans.py

```diff
@@ -25,184 +26,226 @@
 table_builder = SampleSheetTableBuilder()
 
 
-def get_assay_collections(assays, irods_backend):
-    """Return a list of all assay collection names."""
-    return [irods_backend.get_path(a) for a in assays]
-
-
-def get_assay_subcollections(studies, irods_backend):
-    """Return a list of all assay row colletion names."""
-    collections = []
-    for study in studies:
-        try:
-            study_tables = table_builder.get_study_tables(study)
-        except Exception as ex:
-            logger.error(
-                'Study table building exception for "{}" '
-                'in project "{}" ({}): {}'.format(
-                    study.get_display_name(),
-                    study.investigation.project.title,
-                    study.investigation.project.sodar_uuid,
-                    ex,
-                )
+# Local constants
+DELETED = '<DELETED>'
+ERROR = '<ERROR>'
+
+
+class Command(BaseCommand):
+    """Command to find orphans in iRODS collections."""
+
+    help = 'Find orphans in iRODS project collections.'
+
+    def __init__(self):
+        super().__init__()
+        self.irods_backend = get_backend_api('omics_irods')
+
+    def _sort_colls_on_projects(self, collections):
+        """Helper function to sort collections based on project list"""
+        colls_with_project = []
+        colls_no_project = []
+
+        # Create a set of valid project paths based on project UUIDs
+        valid_project_paths = [
+            self.irods_backend.get_path(p)
+            for p in Project.objects.filter(type=PROJECT_TYPE_PROJECT).order_by(
+                'full_title'
+            )
+        ]
+
+        for coll in collections:
+            uuid_prefix = (
```

Comment:
Minor note but this is not a "UUID prefix", it's the entire UUID. It had me confused there for a while :)